.github/workflows/builder_node_slsa3.yml

```diff
@@ -0,0 +1,388 @@
+    runs-on: ubuntu-latest
+    needs: [privacy-check, detect-env, rng]
+    steps:
+      - name: Generate builder
+        id: generate
+        uses: slsa-framework/slsa-github-generator/.github/actions/generate-builder@e3220805577deb9d193f64e519abcb3b50851df5
+        with:
+          repository: "${{ needs.detect-env.outputs.repository }}"
+          ref: "${{ needs.detect-env.outputs.ref }}"
+          go-version: 1.18
+          # Note: This must be the non-randomized binary name, so that it can be downloaded from the release assets.
+          binary: "${{ env.BUILDER_BINARY }}"
+          compile-builder: "${{ inputs.compile-builder }}"
+          directory: "${{ env.BUILDER_DIR }}"
+
+      - name: Upload builder
+        uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # tag=v3.1.0
+        with:
+          name: "${{ env.BUILDER_BINARY }}-${{ needs.rng.outputs.value }}"
+          path: "${{ env.BUILDER_BINARY }}"
+          if-no-files-found: error
+          retention-days: 5
+
+  ###################################################################
+  #                                                                 #
+  #                       Build the project                         #
+  #                                                                 #
+  ###################################################################
+  build:
+    outputs:
+      node-tarball-sha256: ${{ steps.upload.outputs.sha256 }}
+      node-tarball-name: ${{ steps.tarball.outputs.filename }}
+    runs-on: ubuntu-latest
+    needs: [privacy-check, builder, rng]
+    steps:
+      - name: Checkout the Node repository
+        uses: slsa-framework/slsa-github-generator/.github/actions/checkout-node # TODO: set a pin or use at head?
+        with:
+          node-version: ${{ inputs.node-version }}
+          scope: ${{ inputs.scope }}
+          token: ${{ inputs.token }}
+          always-auth: ${{ inputs.always-auth }}
+          registry-url: ${{ inputs.registry-url }}
+          cache: 'npm'
+
+      - name: Download builder
+        uses: slsa-framework/slsa-github-generator/.github/actions/secure-download-artifact@e3220805577deb9d193f64e519abcb3b50851df5
+        with:
+          name: "${{ env.BUILDER_BINARY }}-${{ needs.rng.outputs.value }}"
+          path: "${{ env.BUILDER_BINARY }}"
+          sha256: "${{ needs.builder.outputs.node-builder-sha256 }}"
+          set-executable: true
+
+      - name: Download dependencies
+        env: 
+          UNTRUSTED_CI_ARGUMENTS: "${{ inputs.ci-arguments }}"
+          UNTRUSTED_DIR: ${{ inputs.directory }}"
+        run: |
+          set -euo pipefail
+
+          # npm ci <args>
+          ./"$BUILDER_BINARY" ci \
+            --ci-arguments "$UNTRUSTED_CI_ARGUMENTS" \
+            --directory "$UNTRUSTED_DIR"
+
+      # TODO(hermeticity) OS-level.
+      # - name: Disable hermeticity
+      # uses: slsa/hermeticity@xxx
+      #   with:
+      #     to-state: enabled
```

Comment:
nit: Maybe we can omit this comment?

Reply:
I shortened it a to a single line. Unless we think it would be done in the builder. Let me know.